packages/sdk/src/lib/dataProtectorSharing/setProtectedDataForSale.ts

```diff
@@ -38,116 +39,68 @@ export const setProtectedDataForSale = async ({
+    sharingContract,
+    protectedDataAddress: vProtectedDataAddress,
   });
 
-  if (!protectedData) {
-    throw new ErrorWithData(
-      'This protected data does not exist in the subgraph.',
-      { protectedDataAddress }
-    );
-  }
-
-  if (protectedData.owner.id !== DEFAULT_SHARING_CONTRACT_ADDRESS) {
-    throw new ErrorWithData(
-      'This protected data is not owned by the sharing contract, hence a sharing-related method cannot be called.',
-      {
-        protectedDataAddress,
-        currentOwnerAddress: protectedData.owner.id,
-      }
-    );
-  }
-
-  if (protectedData.collection?.owner?.id !== userAddress) {
-    throw new ErrorWithData(
-      'This protected data is not part of a collection owned by the user.',
-      {
-        protectedDataAddress,
-        currentCollectionOwnerAddress: protectedData.collection?.owner?.id,
-      }
-    );
-  }
-
-  const hasActiveRentals = protectedData.rentals.length > 0;
+  await onlyCollectionOperator({
+    sharingContract,
+    collectionTokenId,
+    userAddress,
+  });
+  await onlyProtectedDataNotRented({
+    sharingContract,
+    protectedDataAddress: vProtectedDataAddress,
+  });
 
-  if (hasActiveRentals) {
-    throw new ErrorWithData('This protected data has active rentals.', {
-      protectedDataAddress,
-      activeRentalsCount: protectedData.rentals.length,
+  try {
+    const includedInSubscription = await isIncludedInSubscription({
+      sharingContract,
+      protectedDataAddress: vProtectedDataAddress,
     });
-  }
-
-  if (protectedData.isRentable && protectedData.isIncludedInSubscription) {
-    throw new ErrorWithData(
-      'This protected data is currently for rent and included in your subscription. First call removeProtectedDataFromRenting() and removeProtectedDataFromSubscription()',
-      {
-        protectedDataAddress,
-      }
-    );
-  }
-
-  if (protectedData.isRentable) {
-    throw new ErrorWithData(
-      'This protected data is currently for rent. First call removeProtectedDataFromRenting()',
-      {
-        protectedDataAddress,
-      }
-    );
-  }
-
-  if (protectedData.isIncludedInSubscription) {
-    // TODO: Create removeProtectedDataFromSubscription() method
-    throw new ErrorWithData(
-      'This protected data is currently included in your subscription. First call removeProtectedDataFromSubscription()',
-      {
-        protectedDataAddress,
-      }
-    );
-  }
-
-  if (protectedData.isForSale) {
-    throw new ErrorWithData('This protected data is already for sale.', {
-      protectedDataAddress,
+    if (includedInSubscription) {
+      // TODO: Create removeProtectedDataFromSubscription() method
+      throw new ErrorWithData(
+        'This protected data is currently included in your subscription. First call removeProtectedDataFromSubscription()',
+        {
+          protectedDataAddress,
+        }
+      );
+    }
+
+    const rentingParams = await getRentingParams({
```

Comment:
Behind this `getRentingParams()` there is a second call to the smart-contract for the same protected data, we could avoid it. (Already done when calling `getCollectionForProtectedData()` at the beginning.)